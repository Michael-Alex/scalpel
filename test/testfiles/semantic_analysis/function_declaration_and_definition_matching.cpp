void f(const int);
void f(volatile int)
{
}

void g(const volatile int);
void g(int)
{
}

void i(int* volatile const);
void i(int*)
{
}

void h(const double* const);
void h(const double ds[5])
{
}

