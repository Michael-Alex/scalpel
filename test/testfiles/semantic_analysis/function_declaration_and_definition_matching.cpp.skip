void f(const int);
void f(volatile int)
{
}

void g(const volatile int);
void g(int)
{
}

void h(int* volatile const);
void h(int*)
{
}

void i(const double* const);
void i(const double ds[5])
{
}

