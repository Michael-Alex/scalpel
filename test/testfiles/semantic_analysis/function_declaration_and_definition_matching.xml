<namespace id="n0">
 <simple_function name="f">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter>
    <type>
     <const>
      <fundamental_type type="int"/>
     </const>
    </type>
   </parameter>
  </parameters>
 </simple_function>
 <simple_function name="g">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter>
    <type>
     <volatile>
      <const>
       <fundamental_type type="int"/>
      </const>
     </volatile>
    </type>
   </parameter>
  </parameters>
 </simple_function>
 <simple_function name="i">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter>
    <type>
     <const>
      <volatile>
       <pointer>
        <fundamental_type type="int"/>
       </pointer>
      </volatile>
     </const>
    </type>
   </parameter>
  </parameters>
 </simple_function>
 <simple_function name="h">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter>
    <type>
     <const>
      <pointer>
       <const>
        <fundamental_type type="double"/>
       </const>
      </pointer>
     </const>
    </type>
   </parameter>
  </parameters>
 </simple_function>
</namespace>
