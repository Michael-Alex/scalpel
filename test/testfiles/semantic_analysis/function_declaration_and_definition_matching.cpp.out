<namespace id="0">
 <simple_function id="0" name="f">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter>
    <type>
     <cv_qualified_type const="true">
      <fundamental_type type="int"/>
     </cv_qualified_type>
    </type>
   </parameter>
  </parameters>
 </simple_function>
 <simple_function id="1" name="g">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter>
    <type>
     <cv_qualified_type const="true" volatile="true">
      <fundamental_type type="int"/>
     </cv_qualified_type>
    </type>
   </parameter>
  </parameters>
 </simple_function>
 <simple_function id="2" name="h">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter>
    <type>
     <cv_qualified_type const="true" volatile="true">
      <pointer>
       <fundamental_type type="int"/>
      </pointer>
     </cv_qualified_type>
    </type>
   </parameter>
  </parameters>
 </simple_function>
 <simple_function id="3" name="i">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter>
    <type>
     <cv_qualified_type const="true">
      <pointer>
       <cv_qualified_type const="true">
        <fundamental_type type="double"/>
       </cv_qualified_type>
      </pointer>
     </cv_qualified_type>
    </type>
   </parameter>
  </parameters>
 </simple_function>
</namespace>
